dds/DCPS/InstanceState.cpp

```diff
@@ -68,8 +71,13 @@ void InstanceState::sample_info(DDS::SampleInfo& si, const ReceivedDataElement*
     static_cast<CORBA::Long>(no_writers_generation_count_);
   si.source_timestamp = de->source_timestamp_;
   si.instance_handle = handle_;
-  RcHandle<DomainParticipantImpl> participant = reader_->participant_servant_.lock();
-  si.publication_handle = participant ? participant->id_to_handle(de->pub_) : 0;
+  RcHandle<DataReaderImpl> reader = reader_.lock();
+  if (reader) {
+    RcHandle<DomainParticipantImpl> participant = reader->participant_servant_.lock();
+    si.publication_handle = participant ? participant->id_to_handle(de->pub_) : 0;
+  } else {
+    si.publication_handle = 0;
```

Comment:
```suggestion
    si.publication_handle = DDS::HANDLE_NIL;
```
Yes, it was wrong before...